src/ConcreteOperations/intersection.jl

```diff
@@ -426,6 +435,37 @@ intersection(H::AbstractHyperrectangle, X::Interval) = _intersection_interval(X,
 intersection(X::Interval, S::AbstractSingleton) = _intersection_singleton(S, X)
 intersection(S::AbstractSingleton, X::Interval) = _intersection_singleton(S, X)
 
+# special case of an axis-aligned half-space and a hyperrectangular set
```

Comment:
fwiw this is an "internal optimization", i'm not sure if we want a new docstring for the method (i suggest not to, but i'm fine adding it too)